src/mongo/client/dbclient_rs.cpp

```diff
@@ -417,10 +418,23 @@ void DBClientReplicaSet::_auth(const BSONObj& params) {
                 break;
             }
 
-            conn->auth(params);
+            BSONObj interposedParams;
 
+            // Force usage of SCRAM-SHA-1 if host has maxWireVersion > 3.
```

Comment:
Given that this code doesn't in some sense really belong here but is here for pragmatic reasons, I think a more elaborated comment is called for.